nav2_rrtconnect_planner/src/rrt_connect.cpp

```diff
@@ -0,0 +1,269 @@
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Shivang Patel
+ *
+ * Reference tutorial:
+ * https://ros-planning.github.io/navigation2/tutorials/docs/writing_new_nav2planner_plugin.html
+ *********************************************************************/
+#include <ompl/base/spaces/SE2StateSpace.h>
+#include <ompl/geometric/planners/rrt/RRTConnect.h>
+#include <ompl/util/PPM.h>
+#include <ompl/tools/config/MagicConstants.h>
+
+#include <iostream>
+#include <fstream>
+#include <utility>
+
+#include "nav2_util/node_utils.hpp"
+#include "nav2_rrtconnect_planner/rrt_connect.hpp"
+
+namespace nav2_rrtconnect_planner
+{
+
+RRTConnect::RRTConnect()
+: tf_(nullptr), costmap_(nullptr), ss_(nullptr), ompl_state_space_(nullptr), bounds_(nullptr)
+{
+
+}
+
+void RRTConnect::configure(
+  rclcpp_lifecycle::LifecycleNode::SharedPtr parent,
+  std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros)
+{
+
+  node_ = parent;
+  tf_ = tf;
+  name_ = name;
+  costmap_ = costmap_ros->getCostmap();
+  global_frame_ = costmap_ros->getGlobalFrameID();
+
+  ompl_initialized_ = false;
+
+  RCLCPP_INFO(
+    node_->get_logger(), "Configuring plugin %s of type NavfnPlanner",
+    name_.c_str());
+
+  // Parameter initialization
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".solve_time",
+    rclcpp::ParameterValue(1.0));
+  node_->get_parameter(name_ + ".solve_time", solve_time_);
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".max_trials",
+    rclcpp::ParameterValue(4));
+  node_->get_parameter(name_ + ".max_trials", max_trials_);
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".collision_checking_resolution", rclcpp::ParameterValue(
+      0.001));
+  node_->get_parameter(name_ + ".collision_checking_resolution", collision_checking_resolution_);
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".allow_unknown", rclcpp::ParameterValue(
+      true));
+  node_->get_parameter(name_ + ".allow_unknown", allow_unknown_);
+}
+
+void RRTConnect::cleanup()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "CleaningUp plugin %s of type NavfnPlanner",
+    name_.c_str());
+}
+
+void RRTConnect::activate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Activating plugin %s of type NavfnPlanner",
+    name_.c_str());
+}
+
+void RRTConnect::deactivate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Deactivating plugin %s of type NavfnPlanner",
+    name_.c_str());
+}
+
+void RRTConnect::initializeOMPL()
+{
+  // Defining Bounds
+  bounds_ = new ompl::base::RealVectorBounds(2);
+  bounds_->setLow(0, costmap_->getOriginX());
```

Comment:
My recollection is that you typically set some normalized bounds and transform your configuration space into it afterwards.

Reply:
okay. Let me try this approach.